assets/js/components/wp-dashboard/WPDashboardPopularPages.js

```diff
@@ -39,45 +44,51 @@ import { numFmt } from '../../util';
 const { useSelect } = Data;
 
 const WPDashboardPopularPages = ( { WidgetReportZero, WidgetReportError } ) => {
-	const dateRangeDates = useSelect( ( select ) =>
-		select( CORE_USER ).getDateRangeDates( {
+	const { report, titles, loading, error } = useSelect( ( select ) => {
+		const dateRangeDates = select( CORE_USER ).getDateRangeDates( {
 			compare: true,
 			offsetDays: DATE_RANGE_OFFSET,
-		} )
-	);
+		} );
+		const reportArgs = {
+			...dateRangeDates,
+			metrics: [
+				{
+					expression: 'ga:pageviews',
+					alias: 'Pageviews',
+				},
+			],
+			dimensions: [ 'ga:pagePath' ],
+			orderby: [
+				{
+					fieldName: 'ga:pageviews',
+					sortOrder: 'DESCENDING',
+				},
+			],
+			limit: 5,
+		};
+		const data = {
+			report: select( MODULES_ANALYTICS ).getReport( reportArgs ),
+			error: select( MODULES_ANALYTICS ).getErrorForSelector(
+				'getReport',
+				[ reportArgs ]
+			),
+			loading: true,
+		};
+		const reportLoaded = select(
+			MODULES_ANALYTICS
+		).hasFinishedResolution( 'getReport', [ reportArgs ] );
 
-	const reportArgs = {
-		...dateRangeDates,
-		metrics: [
-			{
-				expression: 'ga:pageviews',
-				alias: 'Pageviews',
-			},
-		],
-		dimensions: [ 'ga:pageTitle', 'ga:pagePath' ],
-		orderby: [
-			{
-				fieldName: 'ga:pageviews',
-				sortOrder: 'DESCENDING',
-			},
-		],
-		limit: 5,
-	};
+		const { startDate, endDate } = dateRangeDates;
+		data.titles = select( MODULES_ANALYTICS ).getPageTitles( data.report, {
+			startDate,
+			endDate,
+		} );
```

Comment:
Why not pass the entire `reportArgs` as second parameter? It's technically unnecessary right now, but in a way more straightforward than manually passing individual pieces of information. The idea is that in certain cases other report arguments than `startDate` and `endDate` might be relevant for `getPageTitles` as well. For example, if a report argument was some flag to include certain results in the response that otherwise would be excluded by default, we would need to set such a flag in the `getPageTitles` request too.

Not strictly required at this point, but due to the above it conceptually makes sense to pass the entire `reportArgs`.